src/core/homepage_elements/html/views.py

```diff
@@ -15,27 +16,32 @@
 def html_settings(request):
     plugin = models.Plugin.objects.get(name='HTML')
 
-    html_block_content = setting_handler.get_plugin_setting(
+    html_content = setting_handler.get_plugin_setting(
         plugin,
         'html_block_content',
         request.journal,
         create=True,
         pretty='HTML Block Content'
     ).value
 
+    form = forms.HTMLForm(
+        initial={
+            'html_content': html_content,
+        }
+    )
+
     if request.POST:
-        html_block_content = request.POST.get('html_block_content')
-        setting_handler.save_plugin_setting(plugin,
-                                            'html_block_content',
-                                            html_block_content,
-                                            request.journal)
-        messages.add_message(request, messages.INFO, 'HTML Block updated.')
+        form = forms.HTMLForm(
+            request.POST,
+        )
+        if form.is_valid():
+            form.save(request.journal)
+            messages.add_message(request, messages.INFO, 'HTML Block updated.')
         return redirect(reverse('home_settings_index'))
```

Comment:
Should this redirect fire even if the form is not valid?

Reply:
Thanks--fixed it.